lib/osf-components/addon/components/search-page/component.ts

```diff
@@ -219,6 +221,10 @@ export default class SearchPage extends Component<SearchArgs> {
                 resourceTypeFilter = Object.values(ResourceTypeFilterValue).join(',');
             }
             filterQueryObject['resourceType'] = resourceTypeFilter;
+            // Add the accessService if we are not serving locally
+            if (!osfURL.includes('localhost')) {
+                filterQueryObject['accessService'] = osfURL; // Only fetch items from the current OSF environment
+            }
```

Comment:
This would work, but I feel like this is perhaps not the best location to include this code. Should we move this environment-dependent code to the adapter of the `index-card-search` and/or other related models?

Reply:
I like this idea! Will move this now